src/Tickets/Commerce/Order_Modifiers/API/Coupons.php

```diff
@@ -0,0 +1,540 @@
+				'type'        => 'object',
+				'required'    => true,
+				'properties'  => [
+					'name'  => [
+						'description' => esc_html__( 'The purchaser name.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'text-field',
+						'required'    => true,
+					],
+					'email' => [
+						'description' => esc_html__( 'The purchaser email.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'email',
+						'required'    => true,
+					],
+				],
+			],
+		];
+
+		switch ( $endpoint ) {
+			case 'apply':
+			case 'remove':
+				return array_merge( $coupon_args, $common_args );
+
+			case 'validate':
+				return $coupon_args;
+
+			case 'create':
+				return [
+					'slug'         => [
+						'description' => esc_html__( 'The coupon slug.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'text-field',
+						'required'    => true,
+					],
+					'display_name' => [
+						'description' => esc_html__( 'The coupon display name.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'text-field',
+						'required'    => true,
+					],
+					'status'       => [
+						'description' => esc_html__( 'The coupon status.', 'event-tickets' ),
+						'type'        => 'string',
+						'enum'        => [ 'active', 'inactive', 'draft' ],
+						'required'    => true,
+					],
+					'amount'       => [
+						'description' => esc_html__( 'The coupon amount.', 'event-tickets' ),
+						'type'        => 'integer',
+						'required'    => true,
+					],
+					'sub_type'     => [
+						'description' => esc_html__( 'The coupon sub type.', 'event-tickets' ),
+						'type'        => 'string',
+						'enum'        => [ 'percent', 'flat' ],
+						'required'    => false,
+						'default'     => 'flat',
+					],
+					'start_time'   => [
+						'description' => esc_html__( 'The coupon start time.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'date-time',
+						'default'     => null,
+					],
+					'end_time'     => [
+						'description' => esc_html__( 'The coupon end time.', 'event-tickets' ),
+						'type'        => 'string',
+						'format'      => 'date-time',
+						'default'     => null,
+					],
+				];
+
+			default:
+				return [];
+		}
+	}
+
+	/**
+	 * Get the purchaser information.
+	 *
+	 * @since TBD
+	 *
+	 * @param Request $request The request object.
+	 *
+	 * @return array
+	 */
+	protected function get_purchaser_information( Request $request ) {
+		$purchaser_data = $request->get_param( 'purchaser_data' );
+
+		[ $first_name, $last_name ] = explode( ' ', $purchaser_data['name'], 2 );
+
+		return [
+			'purchaser_user_id'    => 0,
+			'purchaser_full_name'  => $purchaser_data['name'],
+			'purchaser_first_name' => $first_name ?? $purchaser_data['name'],
+			'purchaser_last_name'  => $last_name ?? '',
+			'purchaser_email'      => sanitize_email( $purchaser_data['email'] ),
```

Comment:
Yea we should always have that. In cases like that, i prefer failing with a warning rather than being silent about. It's crucial information we are missing we should see that somewhere.

Reply:
Similar to the above, this will always be present because it is a required parameter in the API schema.

Reply:
> Yea we should always have that. In cases like that, i prefer failing with a warning rather than being silent about. It's crucial information we are missing we should see that somewhere.

The REST API handles that for us. If it is missing, the API will generate a message and reject the request.